yt_dlp/extractor/freetv.py

```diff
@@ -0,0 +1,186 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    ExtractorError,
+    try_get,
+    urlencode_postdata, int_or_none, str_or_none,
+)
+
+
+class FreeTvBaseIE(InfoExtractor):
+    def _get_api_response(self, content_id, postdata, resource_type):
+        response = self._download_json(
+            'https://www.freetv.com/wordpress/wp-admin/admin-ajax.php',
+            content_id, data=urlencode_postdata(postdata), fatal=False,
+            note='Downloading {} {} JSON'.format(content_id, resource_type))
+
+        if response is False:
+            raise ExtractorError("Couldn't get response", expected=True)
+
+        if response.get('data') is False:
+            raise ExtractorError("Response doesn't contain {} data".format(resource_type), expected=True)
```

Comment:
Why are these `expected=True`? Actually, why are these even needed? This is in effect the same as not setting `fatal=False` above

Reply:
Sometimes the api returns 200 but contains a json like this

```json
{
   "success": false,
   ....
}
```
At the beginning I was checking for `success`, but I thought that it was better to check for `data` since it is what we end up using.

Should be fixed, I've set `_download_json` to be fatal